src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseStudentParticipationRepository.java

```diff
@@ -48,6 +48,19 @@ SELECT max(prr.id)
     Optional<ProgrammingExerciseStudentParticipation> findByIdWithLatestResultAndFeedbacksAndRelatedSubmissions(@Param("participationId") Long participationId,
             @Param("dateTime") ZonedDateTime dateTime);
 
+    @Query("""
+            SELECT DISTINCT p
+            FROM ProgrammingExerciseStudentParticipation p
+                LEFT JOIN FETCH p.results pr
+                LEFT JOIN FETCH p.submissions
+            WHERE p.id = :participationId AND ((pr.assessmentType = 'AUTOMATIC'
+                        OR (pr.completionDate IS NOT NULL
+                            AND (p.exercise.assessmentDueDate IS NULL
+                                OR p.exercise.assessmentDueDate < :#{#dateTime}))) OR pr.id IS NULL)
+             """)
+    Optional<ProgrammingExerciseStudentParticipation> findByIdWithAllResultsAndRelatedSubmissions(@Param("participationId") Long participationId,
```

Comment:
style + `long` instead of `Long`.
Can you also explain why the `OR pr.id IS NULL` is necessary? I can't think of a case where this would be needed.
```suggestion
            WHERE p.id = :participationId
                AND ((pr.assessmentType = 'AUTOMATIC'
                    OR (pr.completionDate IS NOT NULL
                        AND (p.exercise.assessmentDueDate IS NULL
                            OR p.exercise.assessmentDueDate < :dateTime))))
             """)
    Optional<ProgrammingExerciseStudentParticipation> findByIdWithAllResultsAndRelatedSubmissions(@Param("participationId") long participationId,
```

Reply:
Yes the `OR pr.id IS NULL` is necessary so that you can also fetch the participations that don't have a result yet. Without this when starting a new exercise you first need to make a submission and wait for the result to be able to see the commit history